src/pages/ListSwaps/SwapRow/SwapRow.tsx

```diff
@@ -0,0 +1,179 @@
+import {
+  Button,
+  CircularProgress,
+  TableCell,
+  TableRow
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/styles";
+import React, { useEffect, useReducer } from "react";
+import { RouteComponentProps, withRouter } from "react-router-dom";
+import { EmbeddedRepresentationSubEntity } from "../../../../gen/siren";
+import executeAction from "../../../api/executeAction";
+import { Asset, Properties, toMainUnit } from "../../../api/swapTypes";
+import Dialog from "../../../components/Dialog";
+import ExternalLink from "../../../components/ExternalLink";
+import LedgerActionDialogBody from "../LedgerActionDialogBody";
+import SirenActionParametersDialogBody from "../SirenActionParametersDialogBody";
+import SwapStatusIcon from "../SwapStatusIcon";
+import {
+  actionButtonClicked,
+  actionFailed,
+  actionSuccessful,
+  closeLedgerActionDialog,
+  closeSirenParametersDialog,
+  sirenParameterDialogSubmitted
+} from "./events";
+import { ActionExecutionStatus, initialState, reducer } from "./reducer";
+
+interface AssetCellProps {
+  asset: Asset;
+}
+
+function AssetCell({ asset }: AssetCellProps) {
+  return <span>{toMainUnit(asset)}</span>;
+}
+
+const useStyles = makeStyles(() => ({
+  tableRow: {
+    cursor: "pointer"
+  }
+}));
+
+interface SwapRowProps extends RouteComponentProps {
+  swap: EmbeddedRepresentationSubEntity;
+  reload: () => void;
+}
+
+function SwapRow({ swap, history, reload }: SwapRowProps) {
+  const [
+    {
+      state: {
+        actionExecutionStatus,
+        activeLedgerActionDialog,
+        activeSirenParameterDialog
+      },
+      sideEffect
+    },
+    dispatch
+  ] = useReducer(reducer, initialState);
+
+  useEffect(() => {
+    if (!sideEffect) {
+      return;
+    }
+
+    switch (sideEffect.type) {
+      case "reloadData": {
+        reload();
```

Comment:
:thinking: This reload ends up happening too quickly, so the buttons don't change and the user has to refresh manually to see any changes at all. Would it be bad to slow this down?

Reply:
I have answered this question myself: it's not that bad. Eventually on mainnet the UI should be somehow aware of having performed an action even if the actions available have not changed, but for now this is okay.

Reply:
> This reload ends up happening too quickly, so the buttons don't change and the user has to refresh manually to see any changes at all. Would it be bad to slow this down?

Well, this basically really only works for the `Accept` and `Decline` actions because those are the only ones that are atomically coupled to the comit-node state.

> Eventually on mainnet the UI should be somehow aware of having performed an action even if the actions available have not changed, but for now this is okay.

Yes. That is what I've mentioned a couple of times actually. We need to find a way for the UI to workaround the fact that the comit-node doesn't immediately see the transaction we produced. I think the easiest would be if we would save the txID locally.

For Ethereum, we get a txID directly from metamask and for Bitcoin, we should probably extend the dialog to require one from the user in order to "submit" the dialog. Requiring the txID forces the user to actually create a transaction.
We can then filter out action buttons locally on the comit-i side by checking, if we have already a txID for the pair of swap+actions.

Reply:
I created an issue here: https://github.com/comit-network/comit-i/issues/75